### PR TITLE
Support CUDA 11.7

### PR DESCRIPTION
**Pull request by takagi**

### CODE REVIEW

**Comment by takagi** [comment]
Review ready!

**Review by kmaehashi — APPROVED** [review]
LGTM!

**Comment by kmaehashi** [comment]
Running CUDA 11.7 test: https://ci.preferred.jp/cupy.linux.cuda117/103427/

File: .pfnci/schema.yaml [hunk 1]
```diff
@@ -225,6 +230,8 @@ cudnn:
         alias: "11.6"
       "11.6":
         alias: "11.6"
+      "11.7":
+        alias: "11.7"
```
Comment (kmaehashi) [inline .pfnci/schema.yaml:234]: https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/
```
libcudnn8_8.4.0.27-1+cuda10.2_amd64.deb 224MB 2022-04-01 02:25
libcudnn8_8.4.0.27-1+cuda11.6_amd64.deb 402MB 2022-04-01 02:21
libcudnn8_8.4.1.50-1+cuda10.2_amd64.deb 225MB 2022-05-22 20:26
libcudnn8_8.4.1.50-1+cuda11.6_amd64.deb 401MB 2022-05-20 22:34
```

```suggestion
        alias: "11.6"
```

File: .pfnci/matrix.yaml [hunk 2]
```diff
@@ -228,6 +227,35 @@
...
+# CUDA 11.7 | Linux
+# The latest CUDA version matrix is intended to cover the highest supported combination.
+- project: "cupy.linux.cuda117"
+  target: "cuda117"
+  tags: ["@push", "full", "mini"]
+  system: "linux"
+  os: "ubuntu:20.04"
+  cuda: "11.7"
+  rocm: null
+  nccl: "2.11"
```
Comment (kmaehashi) [inline .pfnci/matrix.yaml:239]:
```suggestion
  nccl: null  # TODO use 2.12 once supported
```

**Review by kmaehashi — CHANGES_REQUESTED** [review]

**Comment by kmaehashi** [comment]
regenerated.

**Comment by kmaehashi** [comment]
/test mini

Testing cuda 11.7 https://ci.preferred.jp/cupy.linux.cuda117/103464/

**Comment by kmaehashi** [comment]
cuda116 test failure is known flaky #5813

**Pull request merged**